tests/test_fields.rs

```diff
@@ -133,3 +133,15 @@ fn header_using_crlf() {
          thirsdcc <secondcc@destination.com>"
     );
 }
+
+#[test]
+fn find_all_header_fields() {
+    let email = Email::from_vec(TEST_EMAIL_CRLF.to_string().into_bytes()).unwrap();
+
+    assert!(
+        email.header_field_names()
+            .iter()
+            .all(|name| email.header_field(name).is_some())
+    );
```

Comment:
We also need to check that what's returned by header_field_names is exhaustive (i.e., all field names present in the email are included).

Reply:
Doesn't the assertion do this? I mean, if one `header_field_name` is NOne when asking for `header_field(header_field_name)`, that assert fails, right?

Reply:
The current assertion correctly checks that all names returned by header_field_names() are proper/present field names, but not that these names are the full list of names present in the email. For example, if the header contains "To", "Cc", "From" fields, but due to some bug header_field_names() returns only "To" and "Cc", this test will still pass.